internal/tracker/artifacts.go

```diff
@@ -56,12 +64,10 @@ func (tracker *Tracker) Add(componentName string) error {
 	return nil
 }
 
-func (tracker *Tracker) MarkContainerd(source string) {
-	tracker.Artifacts.Containerd = source
-}
-
 // Save() saves the tracker to file
 func (tracker *Tracker) Save() error {
+	// ensure containerd source is populated with none/distro/docker
+	tracker.Artifacts.Containerd = GetContainerdSource(string(tracker.Artifacts.Containerd))
```

Comment:
this makes me a bit nervous
if someone adds a new value to the enum but forgets to update `GetContainerdSource`, this will all silently work but default to "none"

What would you think about handling "" and "none" as `ContainerdSourceNone`, but default to error in `GetContainerdSource`? that way we enforce only valid values can be written

I'll admit this is a bit paranoid but you know